packages/tinacms/src/auth/TinaCloudProvider.tsx

```diff
@@ -170,6 +175,30 @@ export const TinaCloudProvider = (
     cms.api.tina.setBranch(currentBranch)
   }
 
+  const setupSearch = async (props: CreateClientProps) => {
+    if (cms.api.search) {
+      return
+    }
+    let searchClient
+    // if local and search is configured then we always use the local client
+    // if not local, then determine if search is enabled and use the client from the config
+    if (props.isLocalClient) {
+      searchClient = new LocalSearchClient(cms.api.tina)
+    } else {
+      const hasTinaSearch = Boolean(props.schema.config?.search?.tina)
+      if (hasTinaSearch) {
+        searchClient = new TinaCMSSearchClient(cms.api.tina)
+      } else {
+        searchClient = props.schema.config?.search?.searchClient
+      }
+    }
+
+    if (searchClient) {
+      cms.registerApi('search', searchClient)
+    }
+  }
+  setupSearch(props)
```

Comment:
Should this be in a useEffect? As it, it will be recalled anytime this is rerendered